psyneulink/library/compositions/pytorchmodelcreator.py

```diff
@@ -43,29 +46,34 @@ def __init__(self, composition, device, context=None):
 
         # Instantiate pytorch mechanisms
         for node in set(composition.nodes) - set(composition.get_nodes_by_role(NodeRole.LEARNING)):
-            pytorch_node = PytorchMechanismWrapper(node, self._composition._get_node_index(node), device, context=context)
-            self.component_map[node] = pytorch_node
+            pytorch_node = wrap_mechanism(node, self._composition._get_node_index(node), device, context=context)
+            self.node_map[node] = pytorch_node
             self.nodes.append(pytorch_node)
+            self.components.append(pytorch_node)
 
         # Instantiate pytorch projections
         for projection in composition.projections:
-            if projection.sender.owner in self.component_map and projection.receiver.owner in self.component_map:
-                proj_send = self.component_map[projection.sender.owner]
-                proj_recv = self.component_map[projection.receiver.owner]
-
-                port_idx = projection.sender.owner.output_ports.index(projection.sender)
-                new_proj = PytorchProjectionWrapper(projection, list(self._composition._inner_projections).index(projection), port_idx, device, sender=proj_send, receiver=proj_recv, context=context)
-                proj_send.add_efferent(new_proj)
-                proj_recv.add_afferent(new_proj)
-                self.projection_map[projection] = new_proj
-                self.projections.append(new_proj)
-                self.params.append(new_proj.matrix)
+            if projection.sender.owner in self.node_map and projection.receiver.owner in self.node_map:
+                proj_send = self.node_map[projection.sender.owner]
+                proj_recv = self.node_map[projection.receiver.owner]
+
+                pytorch_proj = PytorchProjectionWrapper(projection, list(self._composition._inner_projections).index(projection), device, sender=proj_send, receiver=proj_recv, context=context)
+                proj_send.add_efferent(pytorch_proj)
+                proj_recv.add_afferent(pytorch_proj)
+                self.projection_map[projection] = pytorch_proj
+                self.projections.append(pytorch_proj)
+                self.components.append(pytorch_proj)
+
+        # Add component params to params list
+        for component in self.components:
+            for param in component._get_pytorch_params():
+                self.params.append(param)
 
         # Setup execution sets
         # 1) Remove all learning-specific nodes
         self.execution_sets = [x - set(composition.get_nodes_by_role(NodeRole.LEARNING)) for x in composition.scheduler.run(context=context)]
         # 2) Convert to pytorchcomponent representation
-        self.execution_sets = [{self.component_map[comp] for comp in s if comp in self.component_map} for s in self.execution_sets]
+        self.execution_sets = [{self.node_map[comp] for comp in s if comp in self.node_map} for s in self.execution_sets]
```

Comment:
Reassignment  makes things difficult to read each step should use a properly named temp variable.